README.md

```diff
@@ -10,48 +10,66 @@
 
 **Status:** In development (not for production use)
 
 ## Purpose
 
 ### What it does
 
-The code upgrader will convert Apostrophe 2 codebases for installable modules *and* full A2 websites so they are *mostly ready* to run Apostrophe 3. This includes:
+This module's features break down into two basic categories: "linting" for compatibility issues you can fix yourself, and "upgrading" code automatically. Since not everything can be upgraded automatically, the linting feature is important for everyone.
+
+#### Linting for compatibility issues
+
+This module's linting feature scans your project for modifications that likely need to be made to be compatible with A3. The lint command will work well with basically all projects and detects many issues. Here are just a few examples of what the linter can detect:
+
+modifications that likely need to be made to be compatible with A3. The lint command will work well with basically all projects and detects many issues. Here are just a few examples of what the linter can detect:
+
+- The need to rename `lib/modules` to `modules`.
+- The need to change `{{ apos.area(...) }}` to `{% area ... %}`.
+- The need to move code from `construct()` to `methods()`, `handlers()`, etc.
+
+Since the linter is very tolerant it is a good candidate for use with nearly all A2 projects that are migrating to A3.
+
+#### Automated upgrading
+
+This module can also carry out some upgrades automatically. While this is a great feature, keep in mind that not every module and project is a good candidate for automated code upgrading as there are an infinite number of ways projects can be structured. The upgrade featureworks best on projects that adhere very closely to the coding style of the official A2 sample projects and documentation.
+
+There are also many needed changes that the upgrade command cannot handle on its own. So a successful upgrade will always involve reviewing the output of the linting feature,  as well as the Apostrophe documentation.
+
+Where possible, the code upgrader will convert Apostrophe 2 codebases for installable modules *and* full A2 websites so they are *mostly ready* to run Apostrophe 3. This includes:
 - Moving modules from `lib/modules` to the `modules` directory.
 - Renaming most project-level Apostrophe core module customization directories to the A3 equivalents.
-- Converting field schemas, columns, and similarly structured features to the A3 "cascade" configuration structure.
-- Converting utility methods such as `addHelper()`, `apiRoute()`, and others to A3 module customization functions.
+- Converting field schemas, columns, and similarly structured features to the A3 "cascade" configuration structure, if the existing module follows the structure of our A2 example projects closely enough.
```

Comment:
I think we should avoid "our." In the original I'd worded it as "the official A2 sample projects" to not speak from a place of ownership.